### ISSUE TITLE
[Bug] Rule “prefer-readonly” does not handle destructing assignment correctly

### ISSUE THREAD

**Issue by EFanZh** [palantir/tslint#3803]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.8.1
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
class A {
    private a: number = 1;

    public f() {
        [this.a] = [2];
    }
}
```

with `tslint.json` configuration:

```json
{
    "extends": "tslint:latest",
    "rules": {
        "max-classes-per-file": false,
        "no-implicit-dependencies": {
            "options": [
                "dev"
            ]
        },
        "no-submodule-imports": false,
        "prefer-readonly": true,
        "trailing-comma": {
            "options": [
                {
                    "multiline": "never",
                    "singleline": "never"
                }
            ]
        }
    }
}
```

#### Actual behavior

TSLint reports: "Private member variable 'a' is never reassigned; mark it as 'readonly'. (prefer-readonly)".

#### Expected behavior

TSLint does not report error.

**Comment by bowenni** [comment]
According to https://palantir.github.io/tslint/rules/prefer-readonly the `prefer-readonly` rule only checks if the private variable is modified in the constructor so this is working as intended?

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
